docs/tutorial/dataset-definition-concepts.md

```diff
@@ -267,12 +271,14 @@ from ehrql.tables.beta.core import patients, medications
 
 Next, we create a dataset object.
 Don't worry about exactly what this line does!
+# I'd like to know what this is here. I don't think it is explained anywhere
+# e.g. to this we can add the patient series (variables) we are interested in. These form a patient frame that can then be filtered to the population of interest.
```

Comment:
nit:

This and the prior sentences — which weren't your work! — need rewriting.

We do actually explain a little of `Dataset()` above now, which may have not been the case at the time those sentences were written.

> A dataset is a table containing one row per patient, and one column per feature of interest…

I agree that we should probably have something add something along the lines of your suggestion. That is, something like the following points that you've made:

* `Dataset()` represents our output dataset. 
* We add patient "features" (what we've termed variables in ehrQL) as columns to it, by assigning to attributes of the dataset, that is, `dataset.age = … — these have one row per patient
* We define our output dataset's population of interest by using the `define_population()` method on the dataset.
* In ehrQL, methods and attributes are both accessed via Python's "dot" notation.

(Markdown uses `#` as a heading, so this will appear in text, unlike many languages where `#` indicates a comment[^1].)

[^1]: Here's a couple of ways you could actually add comments in Markdown: https://stackoverflow.com/questions/4823468/comments-in-markdown

Reply:
thought:

> e.g. to this we can add the patient series (variables) we are interested in. These form a patient frame that can then be filtered to the population of interest.

I'm not sure how important this is to clarify, but I'll discuss it anyway. And maybe it needs a clearer distinction on the page.

From my (limited) understanding, there is a distinction between `Dataset` and `PatientFrame`. The internal Python representation of [`Dataset`](https://github.com/opensafely-core/ehrql/blob/7eb6bd3f7f66b00d03832c674cc8cccdb259e4b9/ehrql/query_language.py#L39) *is not* a [`PatientFrame`](https://github.com/opensafely-core/ehrql/blob/7eb6bd3f7f66b00d03832c674cc8cccdb259e4b9/ehrql/query_language.py#L961
), nor related to `PatientFrame` by class inheritance.

So, things you can do with a `PatientFrame`, like  `.exists_for_patient()`, you cannot do with a `Dataset`.

:exclamation: **However, `Dataset` does represent a *similar kind of table* as you'd get in a `PatientFrame`**.

Though, `Dataset` is a one row per patient table that you're constructing for the purpose of generating a final output table only. The only filtering operation you can do is to define the population to include with `define_population()`.

I will tag @evansd to correct me if I'm wrong here though. I feel like it's maybe a point of confusion that we should help users to understand, so I need to be correct in what I'm saying too! :smile:

Reply:
Thanks, I didn't know about the comment feature!

I came across [this reference](https://docs.opensafely.org/ehrql/reference/language/#Dataset) for `Dataset`, that I missed and think would have been enough of an explanation. Perhaps just a link to that for now?

Reply:
Yes, that reference basically explains what we've said here. A link to that would be sufficient, I agree :+1: